lib/c-api/src/wasm_c_api/wasmer.rs

```diff
@@ -1,9 +1,35 @@
 //! Wasmer-specific extensions to the Wasm C API.
 
-use crate::wasm_c_api::instance::wasm_instance_t;
+use super::instance::wasm_instance_t;
+use super::module::wasm_module_t;
+use super::types::wasm_name_t;
 use std::ffi::c_void;
+use std::str;
 
 #[no_mangle]
 pub unsafe extern "C" fn wasm_instance_get_vmctx_ptr(instance: &wasm_instance_t) -> *mut c_void {
     instance.inner.vmctx_ptr() as _
 }
+
+#[no_mangle]
+pub unsafe extern "C" fn wasm_module_name(module: &wasm_module_t, out: &mut wasm_name_t) {
+    let name = match module.inner.name() {
+        Some(name) => name,
+        None => return,
+    };
+
+    *out = name.as_bytes().to_vec().into();
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn wasm_module_set_name(module: &wasm_module_t, name: &wasm_name_t) -> bool {
+    let name = match name.into_slice() {
+        Some(name) => match str::from_utf8(name) {
+            Ok(name) => name,
+            Err(_) => return false, // not ideal!
```

Comment:
sometimes what I do for cases like this is to make a private inner function which just returns Option<T> so you can use c_try internally and then do all the logic there and then convert it to whatever return type you actually need...

You can even do it with an anonymous function too

Reply:
I noticed, but I thought this use case was too small to deserve an inner function :-).